packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -26,9 +27,27 @@ import kotlinx.coroutines.sync.withLock
 class Realm private constructor(configuration: RealmConfiguration, dbPointer: NativePointer) :
     BaseRealm(configuration, dbPointer) {
 
-    private val writer: SuspendableWriter = SuspendableWriter(configuration)
+    private val writer: SuspendableWriter = SuspendableWriter(this)
     private val realmPointerMutex = Mutex()
 
+    /**
+     * FIXME Update docs when design is settled
+     * Returns an ID identifying any Realm data at the point in time this method is called.
+     *
+     * This is done by pairing a reference to the public Realm instance alongside the current `dbPointer`.
+     * For live Realms, the `dbPointer` point to a underlying live SharedRealm that might mutate. For frozen
+     * Realms the `dbPointer` points to a frozen SharedRealm that is guaranteed to remain the same, even if
+     * the public Realm advance to a later version.
+     *
+     * The public Realm instance part of the ID can also mutate, so care must be taken if any methods on that
+     * Realm is used.
+     */
+    val updateableRealm: kotlinx.atomicfu.AtomicRef<RealmReference> = kotlinx.atomicfu.atomic<RealmReference>(
+        RealmReference(this, dbPointer)
+    )
+    override val realm: RealmReference
```

Comment:
I have made it a `var`. It could technically be a `val` in `BaseRealm` and then be overwritten as a `var` in `Realm`, but I guess that would still cause confusion if we do generic things on `BaseRealm` over the updates.